### PR TITLE
CSPL-836 Mitigate vulnerabilities from github.com/coreos/prometheus-operator/pkg/client/versioned/scheme-v0.34.0 library

### PR DESCRIPTION
**Pull request by vebken-splunk**
Upgrade prometheus-operator from v0.34.0 to v0.38.3 to address vulnerabilities CVE-2020-8565, CVE-2019-11253, CVE-2019-11254, CVE-2020-14040, and CVE-2020-15113

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,15 +3,15 @@ module github.com/splunk/splunk-operator
 go 1.13
 
 require (
-	github.com/aws/aws-sdk-go v1.17.7
+	github.com/aws/aws-sdk-go v1.30.12
```
Comment (smohan-splunk) [inline go.mod:6]: Do we need this line as this is already being covered as part of https://github.com/splunk/splunk-operator/pull/267/files ?

File: go.mod [hunk 2]
```diff
@@ -3,15 +3,15 @@ module github.com/splunk/splunk-operator
...
+	github.com/coreos/prometheus-operator v0.38.3 // indirect
```
Comment (smohan-splunk) [inline go.mod:7]: Question similar to aws-sdk-go [discussion](https://github.com/splunk/splunk-operator/pull/267#discussion_r585112183) .. does updating to latest version  of prometheus operator cause build failures or other problems?

Comment (vebken-splunk) [reply go.mod:7]: Yes, upgrading to the latest version of prometheus operator causes dependency issues and build issues. v0.38.3 is the latest version that will work without further changes such as upgrading the operator sdk version.

File: go.mod (hunk 1 above)
Comment (vebken-splunk) [reply go.mod:6]: This version is updated automatically by requiring prometheus-operator v0.38.3 so it is necessary for this standalone fix. When merging this and CSPL-841 we will need to ensure that the latest version of the awk-sdk-go library is retained (depending on the order the tickets are merged in).

**Review by smohan-splunk — APPROVED** [review]

**Review by akondur — APPROVED** [review]

**Pull request merged**